receiver/awsxrayreceiver/internal/util/util.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"bytes"
+
+	"go.uber.org/zap"
+)
+
+// Reference for this port:
+// https://github.com/aws/aws-xray-daemon/blob/master/pkg/util/util.go
+
+// ProtocolSeparator is the character used to split the header and body in an
+// X-Ray segment
+const ProtocolSeparator = "\n"
+
+// SplitHeaderBody separates header and body of buf using provided separator sep, and stores in returnByte.
+func SplitHeaderBody(log *zap.Logger, buf, sep *[]byte, returnByte *[][]byte) [][]byte {
+	if buf == nil {
+		log.Error("buffer to split is nil")
```

Comment:
How about returning err instead of just logging? Can leave the decision of whether to log or not to a higher level.